src/main/java/com/friendly_machines/intellij/plugins/ideanative2debugger/impl/DebugProcess.java

```diff
@@ -257,102 +257,95 @@ private SuspendContext generateSuspendContext(List<Object> threads, String curre
         return new SuspendContext(this, stacks.toArray(new ExecutionStack[0]), activeStackId);
     }
 
-    public List<Map<String, Object>> getVariables(String threadId, String frameId) throws GdbMiOperationException, ClassCastException {
-        ArrayList<Map<String, Object>> result = new ArrayList<>();
+    public List<Map<String, ?>> getVariables(String threadId, String frameId) throws GdbMiOperationException, ClassCastException, IOException, InterruptedException {
         // TODO: --simple-values and find stuff yourself.
-        var q = gdbCall("-stack-list-variables", new String[]{"--thread", threadId, "--frame", frameId, "--all-values"}, new String[]{});
-        if (q.containsKey("variables")) {
-            try {
-                List<?> variables = (List<?>) q.get("variables");
-                for (Object variable1 : variables) {
-                    @SuppressWarnings("unchecked")
-                    var variable = (Map<String, Object>) variable1;
-                    result.add(variable);
-                }
-            } catch (ClassCastException e) {
-                e.printStackTrace();
+        var q = gdbCall("-stack-list-variables", List.of("--thread", threadId, "--frame", frameId, "--all-values"), Collections.emptyList());
+        try {
+            var variables = (Collection<Map<String, ?>>) q.get("variables");
+            if (variables == null) {
+                return Collections.emptyList();
             }
+            // pucgenie: Why even copy?
+            return new ArrayList<Map<String, ?>>(variables);
+        } catch (ClassCastException e) {
+            e.printStackTrace();
+            throw new RuntimeException(e);
         }
-
-        return result;
     }
 
-    public List<Map<String, Object>> getFrames(String threadId) throws GdbMiOperationException, ClassCastException {
-        List<Map<String, Object>> result = new ArrayList<>();
-        var q = gdbCall("-stack-list-frames", new String[]{"--thread", threadId}, new String[0]);
-        if (q.containsKey("stack")) {
-            try {
-                List<?> stack = (List<?>) q.get("stack");
-                for (Object frame1 : stack) {
-                    @SuppressWarnings("unchecked")
-                    Map.Entry<String, Object> frame = (Map.Entry<String, Object>) frame1;
-                    if ("frame".equals(frame.getKey())) {
-                        @SuppressWarnings("unchecked")
-                        var item = (Map<String, Object>) frame.getValue();
-                        result.add(item);
-                    }
+    public List<Map<String, ?>> getFrames(String threadId) throws GdbMiOperationException, ClassCastException, IOException, InterruptedException {
+        var q = gdbCall("-stack-list-frames", List.of("--thread", threadId), Collections.emptyList());
+        final var result = new ArrayList<Map<String, ?>>();
+
+        var stack = (Collection<Map.Entry<String, Map<String, ?>>>) q.get("stack");
```

Comment:
`null` can't even be returned by `GdbMiProducer` parsers (not allowed in Gdb/Mi protocol).

Maybe make `GdbMiProducer.parseValue`'s return type @NotNull to make this obvious.